internal/service/accesslistapikey/data_source_accesslist_api_keys.go

```diff
@@ -68,27 +70,44 @@ func PluralDataSource() *schema.Resource {
 	}
 }
 
-func dataSourceMongoDBAtlasAccessListAPIKeysRead(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
-	// Get client connection.
-	conn := meta.(*config.MongoDBClient).Atlas
-	options := &matlas.ListOptions{
-		PageNum:      d.Get("page_num").(int),
-		ItemsPerPage: d.Get("items_per_page").(int),
-	}
-
+func dataSourcePluralRead(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
+	connV2 := meta.(*config.MongoDBClient).AtlasV2
 	orgID := d.Get("org_id").(string)
 	apiKeyID := d.Get("api_key_id").(string)
-
-	accessListAPIKeys, _, err := conn.AccessListAPIKeys.List(ctx, orgID, apiKeyID, options)
+	params := &admin.ListApiKeyAccessListsEntriesApiParams{
+		PageNum:      pointy.Int(d.Get("page_num").(int)),
+		ItemsPerPage: pointy.Int(d.Get("items_per_page").(int)),
+		OrgId:        orgID,
+		ApiUserId:    apiKeyID,
+	}
+	accessListAPIKeys, _, err := connV2.ProgrammaticAPIKeysApi.ListApiKeyAccessListsEntriesWithParams(ctx, params).Execute()
 	if err != nil {
 		return diag.FromErr(fmt.Errorf("error getting access list api keys information: %s", err))
 	}
 
-	if err := d.Set("results", flattenAccessListAPIKeys(ctx, conn, orgID, accessListAPIKeys.Results)); err != nil {
+	if err := d.Set("results", flattenAccessListAPIKeys(ctx, orgID, accessListAPIKeys.GetResults())); err != nil {
 		return diag.FromErr(fmt.Errorf("error setting `results`: %s", err))
 	}
 
 	d.SetId(id.UniqueId())
 
 	return nil
 }
+
+func flattenAccessListAPIKeys(ctx context.Context, orgID string, list []admin.UserAccessList) []map[string]any {
+	var results []map[string]any
+	if len(list) > 0 {
```

Comment:
[nit] removing one if level
```suggestion
	if len(list) == 0 {
	  return results
	}
```

Reply:
thx, changed